src/openzaak/components/zaken/api/validators.py

```diff
@@ -587,3 +589,34 @@ def __call__(self, attrs, serializer):
             raise serializers.ValidationError(
                 self.message.format(spec=spec), code=self.code
             )
+
+
+class AuthContextMandateValidator:
+    code = "required"
+    message = _("The mandate context is required when a representee is specified.")
+
+    def __call__(self, attrs: dict):
+        if attrs.get("representee") and not attrs.get("mandate"):
+            raise serializers.ValidationError(
+                {"mandate": ErrorDetail(self.message, code=self.code)}
+            )
+
+
+class RolIndicatieMachtigingValidator:
+    code = "indicatie-machtiging-invalid"
+    message = _(
+        "`indicatieMachtiging` must be set to 'gemachtigde' when a representee is specified"
+    )
+
+    def __call__(self, attrs: dict):
+        indicatie_machtiging = attrs.get("indicatie_machtiging")
+        authenticatie_context = attrs.get("authenticatie_context")
+
+        if not authenticatie_context:
+            return
```

Comment:
Why do you ignore when no `authenticatie_context` but not above?



> if ``representee`` is provided in ``authenticatieContext``, then:
> * ``indicatieMachtiging`` **MUST** be set to "gemachtigde"
> * ``mandate`` **MUST** be provided in the ``authenticatieContext``

Reply:
Actually I get it. `representee` and `mandate` are both in `authenticatie_context`

Reply:
Yes! The idea is if you represent someone you should have some mandate for it, but we just check that it exists